OSDOCS-20088: Compliance Operator 1.9.1 z-stream release notes (#113004)
openshift/openshift-docs: modules/compliance-rn-1-9-1.adoc

// Module included in the following assemblies:
//
// * security/compliance_operator/compliance-operator-release-notes.adoc

:_mod-docs-content-type: REFERENCE
[id="compliance-operator-release-notes-1-9-1_{context}"]
= Release notes for OpenShift Compliance Operator 1.9.1

[role="_abstract"]
Release notes for OpenShift Compliance Operator 1.9.1.

The following Red Hat Security Advisory (RHSA) is available for the OpenShift Compliance Operator 1.9.1:

* link:https://access.redhat.com/errata/RHSA-2026:xxxxx[RHSA-2026:xxxxx - OpenShift Compliance Operator 1.9.1 bug fix and enhancement update]

[id="compliance-operator-1-9-1-bug-fixes_{context}"]
== Bug fixes

> * Before this release, when you created a `TailoredProfile` object that extended a base profile, you could only enable rules that were not available in the XCCDF groups enabled in the base profile. With this release, `TailoredProfile` objects can now enable any rule available. For more information, see (link:https://redhat.atlassian.net/browse/CMP-4283[CMP-4283]).

Reviewer: ```suggestion
* Before this release, when you created a `TailoredProfile` object that extended a base profile, you could only enable rules that were available in the XCCDF groups enabled in the base profile. With this release, `TailoredProfile` objects can now enable any rule available. For more information, see (link:https://redhat.atlassian.net/browse/CMP-4283[CMP-4283]).
```
`enable` is used twice in a single sentence, don't know how to improve that though.

Author: This is done, thank you. I simply deleted the second "enabled".

// Module included in the following assemblies:
//
// * security/compliance_operator/compliance-operator-release-notes.adoc

:_mod-docs-content-type: REFERENCE
[id="compliance-operator-release-notes-1-9-1_{context}"]
= Release notes for OpenShift Compliance Operator 1.9.1

[role="_abstract"]
Release notes for OpenShift Compliance Operator 1.9.1.

The following Red Hat Security Advisory (RHSA) is available for the OpenShift Compliance Operator 1.9.1:

* link:https://access.redhat.com/errata/RHSA-2026:xxxxx[RHSA-2026:xxxxx - OpenShift Compliance Operator 1.9.1 bug fix and enhancement update]

[id="compliance-operator-1-9-1-new-features-and-enhancements_{context}"]
== New features and enhancements

[id="compliance-operator-1-9-1-bug-fixes_{context}"]
== Bug fixes

> * Before this release, when you created a `TailoredProfile` that extended a base profile, you could not enable default rules that were not available in the base profile. With this release, `TailoredProfile` objects can adopt those default rules. For more information, see (link:https://redhat.atlassian.net/browse/CMP-4283[CMP-4283]).

Reviewer: ```suggestion
* Before this release, when you created a `TailoredProfile` that extended a base profile, you could not enable rules that were not available in the XCCDF groups enabled in the base profile. With this release, `TailoredProfile` objects can now enable any rule available. For more information, see (link:https://redhat.atlassian.net/browse/CMP-4283[CMP-4283]).
```

Alternative avoiding negatives:
```suggestion
* Before this release, when you created a `TailoredProfile` that extended a base profile, you could only enable rules that were available in the XCCDF groups enabled in the base profile. With this release, `TailoredProfile` objects can now enable any rule available. For more information, see (link:https://redhat.atlassian.net/browse/CMP-4283[CMP-4283]).
```

Author: Thank you @yuumasato ! I took your option to get rid of the double negatives. I worried that's truly what dev meant, having you suggest different helped.

> * Before this release, the Compliance Operator rule `ocp4-cis-file-permissions-cni-conf` checked file permissions for every file under `/etc/cni/net.d/`, including the runtime `cni.lock` file, which could cause incorrect fail results. With this release, the rule checks permissions only for CNI configuration files (`.conf`, `.conflist`, `.json`). For more information, see (link:https://redhat.atlassian.net/browse/CMP-4323[CMP-4323]).

Reviewer 2: ```suggestion
* Before this release, the Compliance Operator rule, `ocp4-cis-file-permissions-cni-conf`, checked file permissions for every file under the `/etc/cni/net.d/` directory, including the runtime `cni.lock` file, which could cause incorrect fail results. With this release, the rule checks permissions only for CNI configuration files (`.conf`, `.conflist`, `.json`). For more information, see (link:https://redhat.atlassian.net/browse/CMP-4323[CMP-4323]).
```

Author: Done, thank  you!

* Before this release, the Compliance Operator rule `ocp4-directory-access-var-log-kube-audit` could report a non-compliance status on control plane nodes because the `.lock` and `.terminating` metadata files in `/var/log/kube-apiserver/` were included in the directory permission check. With this release, the rule no longer fails on those files. For more information, see (link:https://redhat.atlassian.net/browse/CMP-4321[CMP-4321]).

* Before this release, the Compliance Operator rule `ocp4-cis-file-permissions-cni-conf` checked file permissions for every file under `/etc/cni/net.d/`, including the runtime `cni.lock` file, which could cause incorrect fail results. With this release, the rule checks permissions only for CNI configuration files (`.conf`, `.conflist`, `.json`). For more information, see (link:https://redhat.atlassian.net/browse/CMP-4323[CMP-4323]).

> 

Reviewer: These 2 issues are still in progress, and I don't think they will make it into the release.

Reviewer 3: Based on what Vincent just said at the office hours, CMP-4323 is still in the game - it is going to be part of the CO 1.9.1. because it is very small change and easy fix

Author: OK! Thank you Anna! I am standing by!

> * Before this release, updated DISA STIG reference URLs caused the profile parser to omit STIG reference annotations on `Rule` custom resources. With this release, the parser recognizes the updated URLs and restores those annotations. For more information, see (link:https://redhat.atlassian.net/browse/CMP-4333[CMP-4333]).

Reviewer 2: Do you need to spell out DISA STIG?

Author: No, it's a highly known security reference among  my readers. It occurs many times in the release notes.

> * Before this release, when a `ProfileBundle` content image changed during parsing, multiple profile parser pods could race to update bundle status and leave the bundle stuck in a non-`VALID` state. With this release, the profile parser deployment uses a `Recreate` update strategy and retries status updates on conflict, so bundles reach `VALID` reliably after content is fixed. For more information, see (link:https://redhat.atlassian.net/browse/CMP-4324[CMP-4324]).

Reviewer 3: I think this is just an internal fix for our flaky e2e tests and does not need to be documented in the release notes. I am not 100% sure though.

Author: Thank you @Anna-Koudelkova I have completely removed that note, your team members agree with you!

> * Before this release, updated NERC-CIP reference URLs in `ComplianceAsCode` caused the profile parser to omit NERC-CIP annotations on `Rule` custom resources. With this release, the parser recognizes the updated URLs and restores those annotations. For more information, see (link:https://redhat.atlassian.net/browse/CMP-4349[CMP-4349]).

Reviewer 4: Thanks James. Small nit:  What do we mean by _ComplianceAsCode_? Is it the repo or the organization we are referring to? Might be helpful if we are a bit more explicit?

Author: @taimurhafeez it seems like the sentence would be more useful if I simply deleted "in ComplianceAsCode", right?

Author: This is done, thank you!
